### PR TITLE
Add support for ListInstanceTypes method

### PR DESCRIPTION
**Pull request by AkashKanni**
This PR adds ListInstanceTypes function to list
all available instance-types (rosa list instance-types) and 
it also verifies the presence of newly enabled instance-types

$ ginkgo --focus 73308 tests/e2e
Ginkgo detected a version mismatch between the Ginkgo CLI and the version of Ginkgo imported by your packages:
  Ginkgo CLI Version:
    2.11.0
  Mismatched package versions found:
    2.17.1 used by e2e

  Ginkgo will continue to attempt to run but you may see errors (including flag
  parsing errors) and should either update your go.mod or your version of the
  Ginkgo CLI to match.

  To install the matching version of the CLI run
    go install github.com/onsi/ginkgo/v2/ginkgo
  from a path that contains a go.mod file.  Alternatively you can use
    go run github.com/onsi/ginkgo/v2/ginkgo
  from a path that contains a go.mod file to invoke the matching version of the
  Ginkgo CLI.

  If you are attempting to test multiple packages that each have a different
  version of the Ginkgo library with a single Ginkgo CLI that is currently
  unsupported.
  
Running Suite: e2e tests suite - /home/akanni/OCP-Repository/rosa/tests/e2e
===========================================================================
Random Seed: 1714993591

Will run 1 of 58 specs
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSStime="2024-05-06T16:36:33+05:30" level=info msg="Running command: rosa list instance-types"
time="2024-05-06T16:36:36+05:30" level=info msg="Get Combining Stdout and Stder is :\nID                 CATEGORY               CPU_CORES  MEMORY  \ndl1.24xlarge       accelerated_computing  96         768.0 GiB\ng4dn.12xlarge      accelerated_computing  48         192.0 GiB\ng4dn.16xlarge      accelerated_computing  64         256.0 GiB\ng4dn.2xlarge       accelerated_computing  8          32.0 GiB\ng4dn.4xlarge       accelerated_computing  16         64.0 GiB\ng4dn.8xlarge       accelerated_computing  32         128.0 GiB\ng4dn.metal         accelerated_computing  96         384.0 GiB\ng4dn.xlarge        accelerated_computing  4          16.0 GiB\ng5.12xlarge        accelerated_computing  48         192.0 GiB\ng5.16xlarge        accelerated_computing  64         256.0 GiB\ng5.24xlarge        accelerated_computing  96         384.0 GiB\ng5.2xlarge         accelerated_computing  8          32.0 GiB\ng5.48xlarge        accelerated_computing  192        768.0 GiB\ng5.4xlarge         accelerated_computing  16         64.0 GiB\ng5.8xlarge         accelerated_computing  32         128.0 GiB\ng5.xlarge          accelerated_computing  4          16.0 GiB\np3.16xlarge        accelerated_computing  64         488.0 GiB\np3.2xlarge         accelerated_computing  8          61.0 GiB\np3.8xlarge         accelerated_computing  32         244.0 GiB\np3dn.24xlarge      accelerated_computing  96         768.0 GiB\np4d.24xlarge       accelerated_computing  96         1.1 TiB\np4de.24xlarge      accelerated_computing  96         1.1 TiB\np5.48xlarge        accelerated_computing  192        2.0 TiB\nt3.2xlarge         burstable              8          32.0 GiB\nt3a.2xlarge        burstable              8          32.0 GiB\nt3a.xlarge         burstable              4          16.0 GiB\nt3.xlarge          burstable              4          16.0 GiB\nc5.12xlarge        compute_optimized      48         96.0 GiB\nc5.18xlarge        compute_optimized      72         144.0 GiB\nc5.24xlarge        compute_optimized      96         192.0 GiB\nc5.2xlarge         compute_optimized      8          16.0 GiB\nc5.4xlarge         compute_optimized      16         32.0 GiB\nc5.9xlarge         compute_optimized      36         72.0 GiB\nc5a.12xlarge       compute_optimized      48         96.0 GiB\nc5a.16xlarge       compute_optimized      64         128.0 GiB\nc5a.24xlarge       compute_optimized      96         192.0 GiB\nc5a.2xlarge        compute_optimized      8          16.0 GiB\nc5a.4xlarge        compute_optimized      16         32.0 GiB\nc5a.8xlarge        compute_optimized      32         64.0 GiB\nc5a.xlarge         compute_optimized      4          8.0 GiB\nc5d.12xlarge       compute_optimized      48         96.0 GiB\nc5d.18xlarge       compute_optimized      72         144.0 GiB\nc5d.24xlarge       compute_optimized      96         192.0 GiB\nc5d.2xlarge        compute_optimized      8          16.0 GiB\nc5d.4xlarge        compute_optimized      16         32.0 GiB\nc5d.9xlarge        compute_optimized      36         72.0 GiB\nc5d.metal          compute_optimized      96         192.0 GiB\nc5d.xlarge         compute_optimized      4          8.0 GiB\nc5.metal           compute_optimized      96         192.0 GiB\nc5n.metal          compute_optimized      72         192.0 GiB\nc5.xlarge          compute_optimized      4          8.0 GiB\nc6a.12xlarge       compute_optimized      48         96.0 GiB\nc6a.16xlarge       compute_optimized      64         128.0 GiB\nc6a.24xlarge       compute_optimized      96         192.0 GiB\nc6a.2xlarge        compute_optimized      8          16.0 GiB\nc6a.32xlarge       compute_optimized      128        256.0 GiB\nc6a.48xlarge       compute_optimized      192        384.0 GiB\nc6a.4xlarge        compute_optimized      16         32.0 GiB\nc6a.8xlarge        compute_optimized      32         64.0 GiB\nc6a.xlarge         compute_optimized      4          8.0 GiB\nc6i.12xlarge       compute_optimized      48         96.0 GiB\nc6i.16xlarge       compute_optimized      64         128.0 GiB\nc6i.24xlarge       compute_optimized      96         192.0 GiB\nc6i.2xlarge        compute_optimized      8          16.0 GiB\nc6i.32xlarge       compute_optimized      128        256.0 GiB\nc6i.4xlarge        compute_optimized      16         32.0 GiB\nc6i.8xlarge        compute_optimized      32         64.0 GiB\nc6id.12xlarge      compute_optimized      48         96.0 GiB\nc6id.16xlarge      compute_optimized      64         128.0 GiB\nc6id.24xlarge      compute_optimized      96         192.0 GiB\nc6id.2xlarge       compute_optimized      8          16.0 GiB\nc6id.32xlarge      compute_optimized      128        256.0 GiB\nc6id.4xlarge       compute_optimized      16         32.0 GiB\nc6id.8xlarge       compute_optimized      32         64.0 GiB\nc6id.metal         compute_optimized      128        256.0 GiB\nc6id.xlarge        compute_optimized      4          8.0 GiB\nc6i.metal          compute_optimized      128        256.0 GiB\nc6in.12xlarge      compute_optimized      48         96.0 GiB\nc6in.16xlarge      compute_optimized      64         128.0 GiB\nc6in.24xlarge      compute_optimized      96         192.0 GiB\nc6in.2xlarge       compute_optimized      8          16.0 GiB\nc6in.32xlarge      compute_optimized      128        256.0 GiB\nc6in.4xlarge       compute_optimized      16         32.0 GiB\nc6in.8xlarge       compute_optimized      32         64.0 GiB\nc6in.xlarge        compute_optimized      4          8.0 GiB\nc6i.xlarge         compute_optimized      4          8.0 GiB\nc7a.12xlarge       compute_optimized      48         96.0 GiB\nc7a.16xlarge       compute_optimized      64         128.0 GiB\nc7a.24xlarge       compute_optimized      96         192.0 GiB\nc7a.2xlarge        compute_optimized      8          16.0 GiB\nc7a.32xlarge       compute_optimized      128        256.0 GiB\nc7a.48xlarge       compute_optimized      192        384.0 GiB\nc7a.4xlarge        compute_optimized      16         32.0 GiB\nc7a.8xlarge        compute_optimized      32         64.0 GiB\nc7a.metal-48xl     compute_optimized      192        384.0 GiB\nc7a.xlarge         compute_optimized      4          8.0 GiB\nc7gd.12xlarge      compute_optimized      48         96.0 GiB\nc7gd.16xlarge      compute_optimized      64         128.0 GiB\nc7gd.2xlarge       compute_optimized      8          16.0 GiB\nc7gd.4xlarge       compute_optimized      16         32.0 GiB\nc7gd.8xlarge       compute_optimized      32         64.0 GiB\nc7gd.xlarge        compute_optimized      4          8.0 GiB\nc7gn.12xlarge      compute_optimized      48         96.0 GiB\nc7gn.16xlarge      compute_optimized      64         128.0 GiB\nc7gn.2xlarge       compute_optimized      8          16.0 GiB\nc7gn.4xlarge       compute_optimized      16         32.0 GiB\nc7gn.8xlarge       compute_optimized      32         64.0 GiB\nc7gn.xlarge        compute_optimized      4          8.0 GiB\nc7i.12xlarge       compute_optimized      48         96.0 GiB\nc7i.16xlarge       compute_optimized      64         128.0 GiB\nc7i.24xlarge       compute_optimized      96         192.0 GiB\nc7i.2xlarge        compute_optimized      8          16.0 GiB\nc7i.48xlarge       compute_optimized      192        384.0 GiB\nc7i.4xlarge        compute_optimized      16         32.0 GiB\nc7i.8xlarge        compute_optimized      32         64.0 GiB\nc7i.metal-24xl     compute_optimized      96         192.0 GiB\nc7i.metal-48xl     compute_optimized      192        384.0 GiB\nc7i.xlarge         compute_optimized      4          8.0 GiB\nhpc6a.48xlarge     compute_optimized      96         384.0 GiB\nhpc6id.32xlarge    compute_optimized      64         1.0 TiB\nhpc7a.12xlarge     compute_optimized      24         768.0 GiB\nhpc7a.24xlarge     compute_optimized      48         768.0 GiB\nhpc7a.48xlarge     compute_optimized      96         768.0 GiB\nhpc7a.96xlarge     compute_optimized      192        768.0 GiB\nm5zn.12xlarge      compute_optimized      48         192.0 GiB\nm5zn.2xlarge       compute_optimized      8          32.0 GiB\nm5zn.3xlarge       compute_optimized      16         48.0 GiB\nm5zn.6xlarge       compute_optimized      32         96.0 GiB\nm5zn.xlarge        compute_optimized      4          16.0 GiB\nm5.12xlarge        general_purpose        48         192.0 GiB\nm5.16xlarge        general_purpose        64         256.0 GiB\nm5.24xlarge        general_purpose        96         384.0 GiB\nm5.2xlarge         general_purpose        8          32.0 GiB\nm5.4xlarge         general_purpose        16         64.0 GiB\nm5.8xlarge         general_purpose        32         128.0 GiB\nm5a.12xlarge       general_purpose        48         192.0 GiB\nm5a.16xlarge       general_purpose        64         256.0 GiB\nm5a.24xlarge       general_purpose        96         384.0 GiB\nm5a.2xlarge        general_purpose        8          32.0 GiB\nm5a.4xlarge        general_purpose        16         64.0 GiB\nm5a.8xlarge        general_purpose        32         128.0 GiB\nm5a.xlarge         general_purpose        4          16.0 GiB\nm5d.metal          general_purpose        96         384.0 GiB\nm5dn.metal         general_purpose        96         384.0 GiB\nm5.metal           general_purpose        96         384.0 GiB\nm5n.metal          general_purpose        96         384.0 GiB\nm5.xlarge          general_purpose        4          16.0 GiB\nm5zn.metal         general_purpose        48         192.0 GiB\nm6a.12xlarge       general_purpose        48         192.0 GiB\nm6a.16xlarge       general_purpose        64         256.0 GiB\nm6a.24xlarge       general_purpose        96         384.0 GiB\nm6a.2xlarge        general_purpose        8          32.0 GiB\nm6a.32xlarge       general_purpose        128        512.0 GiB\nm6a.48xlarge       general_purpose        192        768.0 GiB\nm6a.4xlarge        general_purpose        16         64.0 GiB\nm6a.8xlarge        general_purpose        32         128.0 GiB\nm6a.metal          general_purpose        192        768.0 GiB\nm6a.xlarge         general_purpose        4          16.0 GiB\nm6i.12xlarge       general_purpose        48         192.0 GiB\nm6i.16xlarge       general_purpose        64         256.0 GiB\nm6i.24xlarge       general_purpose        96         384.0 GiB\nm6i.2xlarge        general_purpose        8          32.0 GiB\nm6i.32xlarge       general_purpose        128        512.0 GiB\nm6i.4xlarge        general_purpose        16         64.0 GiB\nm6i.8xlarge        general_purpose        32         128.0 GiB\nm6id.12xlarge      general_purpose        48         192.0 GiB\nm6id.16xlarge      general_purpose        64         256.0 GiB\nm6id.24xlarge      general_purpose        96         384.0 GiB\nm6id.2xlarge       general_purpose        8          32.0 GiB\nm6id.32xlarge      general_purpose        128        512.0 GiB\nm6id.4xlarge       general_purpose        16         64.0 GiB\nm6id.8xlarge       general_purpose        32         128.0 GiB\nm6id.metal         general_purpose        128        512.0 GiB\nm6idn.12xlarge     general_purpose        48         192.0 GiB\nm6idn.16xlarge     general_purpose        64         256.0 GiB\nm6idn.24xlarge     general_purpose        96         384.0 GiB\nm6idn.2xlarge      general_purpose        8          32.0 GiB\nm6idn.32xlarge     general_purpose        128        512.0 GiB\nm6idn.4xlarge      general_purpose        16         64.0 GiB\nm6idn.8xlarge      general_purpose        32         128.0 GiB\nm6idn.xlarge       general_purpose        4          16.0 GiB\nm6id.xlarge        general_purpose        4          16.0 GiB\nm6i.metal          general_purpose        128        512.0 GiB\nm6in.12xlarge      general_purpose        48         192.0 GiB\nm6in.16xlarge      general_purpose        64         256.0 GiB\nm6in.24xlarge      general_purpose        96         384.0 GiB\nm6in.2xlarge       general_purpose        8          32.0 GiB\nm6in.32xlarge      general_purpose        128        512.0 GiB\nm6in.4xlarge       general_purpose        16         64.0 GiB\nm6in.8xlarge       general_purpose        32         128.0 GiB\nm6in.xlarge        general_purpose        4          16.0 GiB\nm6i.xlarge         general_purpose        4          16.0 GiB\nm7a.12xlarge       general_purpose        48         192.0 GiB\nm7a.16xlarge       general_purpose        64         256.0 GiB\nm7a.24xlarge       general_purpose        96         384.0 GiB\nm7a.2xlarge        general_purpose        8          32.0 GiB\nm7a.32xlarge       general_purpose        128        512.0 GiB\nm7a.48xlarge       general_purpose        192        768.0 GiB\nm7a.4xlarge        general_purpose        16         64.0 GiB\nm7a.8xlarge        general_purpose        32         128.0 GiB\nm7a.metal-48xl     general_purpose        192        768.0 GiB\nm7a.xlarge         general_purpose        4          16.0 GiB\nm7gd.12xlarge      general_purpose        48         192.0 GiB\nm7gd.16xlarge      general_purpose        64         256.0 GiB\nm7gd.2xlarge       general_purpose        8          32.0 GiB\nm7gd.4xlarge       general_purpose        16         64.0 GiB\nm7gd.8xlarge       general_purpose        32         128.0 GiB\nm7gd.xlarge        general_purpose        4          16.0 GiB\nm7i.12xlarge       general_purpose        48         192.0 GiB\nm7i.16xlarge       general_purpose        64         256.0 GiB\nm7i.24xlarge       general_purpose        96         384.0 GiB\nm7i.2xlarge        general_purpose        8          32.0 GiB\nm7i.48xlarge       general_purpose        192        768.0 GiB\nm7i.4xlarge        general_purpose        16         64.0 GiB\nm7i.8xlarge        general_purpose        32         128.0 GiB\nm7i-flex.2xlarge   general_purpose        8          32.0 GiB\nm7i-flex.4xlarge   general_purpose        16         64.0 GiB\nm7i-flex.8xlarge   general_purpose        32         128.0 GiB\nm7i-flex.xlarge    general_purpose        4          16.0 GiB\nm7i.metal-24xl     general_purpose        96         384.0 GiB\nm7i.metal-48xl     general_purpose        192        768.0 GiB\nm7i.xlarge         general_purpose        4          16.0 GiB\nr4.16xlarge        memory_optimized       64         488.0 GiB\nr4.2xlarge         memory_optimized       8          61.0 GiB\nr4.4xlarge         memory_optimized       16         122.0 GiB\nr4.8xlarge         memory_optimized       32         244.0 GiB\nr4.xlarge          memory_optimized       4          30.5 GiB\nr5.12xlarge        memory_optimized       48         384.0 GiB\nr5.16xlarge        memory_optimized       64         512.0 GiB\nr5.24xlarge        memory_optimized       96         768.0 GiB\nr5.2xlarge         memory_optimized       8          64.0 GiB\nr5.4xlarge         memory_optimized       16         128.0 GiB\nr5.8xlarge         memory_optimized       32         256.0 GiB\nr5a.12xlarge       memory_optimized       48         384.0 GiB\nr5a.16xlarge       memory_optimized       64         512.0 GiB\nr5a.24xlarge       memory_optimized       96         768.0 GiB\nr5a.2xlarge        memory_optimized       8          64.0 GiB\nr5a.4xlarge        memory_optimized       16         128.0 GiB\nr5a.8xlarge        memory_optimized       32         256.0 GiB\nr5ad.12xlarge      memory_optimized       48         384.0 GiB\nr5ad.16xlarge      memory_optimized       64         512.0 GiB\nr5ad.24xlarge      memory_optimized       96         768.0 GiB\nr5ad.2xlarge       memory_optimized       8          64.0 GiB\nr5ad.4xlarge       memory_optimized       16         128.0 GiB\nr5ad.8xlarge       memory_optimized       32         256.0 GiB\nr5ad.xlarge        memory_optimized       4          32.0 GiB\nr5a.xlarge         memory_optimized       4          32.0 GiB\nr5b.12xlarge       memory_optimized       48         384.0 GiB\nr5b.16xlarge       memory_optimized       64         512.0 GiB\nr5b.24xlarge       memory_optimized       96         768.0 GiB\nr5b.2xlarge        memory_optimized       8          64.0 GiB\nr5b.4xlarge        memory_optimized       16         128.0 GiB\nr5b.8xlarge        memory_optimized       32         256.0 GiB\nr5b.metal          memory_optimized       96         768.0 GiB\nr5b.xlarge         memory_optimized       4          32.0 GiB\nr5d.12xlarge       memory_optimized       48         384.0 GiB\nr5d.16xlarge       memory_optimized       64         512.0 GiB\nr5d.24xlarge       memory_optimized       96         768.0 GiB\nr5d.2xlarge        memory_optimized       8          64.0 GiB\nr5d.4xlarge        memory_optimized       16         128.0 GiB\nr5d.8xlarge        memory_optimized       32         256.0 GiB\nr5d.metal          memory_optimized       96         768.0 GiB\nr5dn.12xlarge      memory_optimized       48         384.0 GiB\nr5dn.16xlarge      memory_optimized       64         512.0 GiB\nr5dn.24xlarge      memory_optimized       96         768.0 GiB\nr5dn.2xlarge       memory_optimized       8          64.0 GiB\nr5dn.4xlarge       memory_optimized       16         128.0 GiB\nr5dn.8xlarge       memory_optimized       32         256.0 GiB\nr5dn.metal         memory_optimized       96         768.0 GiB\nr5dn.xlarge        memory_optimized       4          32.0 GiB\nr5d.xlarge         memory_optimized       4          32.0 GiB\nr5.metal           memory_optimized       96         768.0 GiB\nr5n.12xlarge       memory_optimized       48         384.0 GiB\nr5n.16xlarge       memory_optimized       64         512.0 GiB\nr5n.24xlarge       memory_optimized       96         768.0 GiB\nr5n.2xlarge        memory_optimized       8          64.0 GiB\nr5n.4xlarge        memory_optimized       16         128.0 GiB\nr5n.8xlarge        memory_optimized       32         256.0 GiB\nr5n.metal          memory_optimized       96         768.0 GiB\nr5n.xlarge         memory_optimized       4          32.0 GiB\nr5.xlarge          memory_optimized       4          32.0 GiB\nr6a.12xlarge       memory_optimized       48         384.0 GiB\nr6a.16xlarge       memory_optimized       64         512.0 GiB\nr6a.24xlarge       memory_optimized       96         768.0 GiB\nr6a.2xlarge        memory_optimized       8          64.0 GiB\nr6a.32xlarge       memory_optimized       128        1.0 TiB\nr6a.48xlarge       memory_optimized       192        1.5 TiB\nr6a.4xlarge        memory_optimized       16         128.0 GiB\nr6a.8xlarge        memory_optimized       32         256.0 GiB\nr6a.xlarge         memory_optimized       4          32.0 GiB\nr6i.12xlarge       memory_optimized       48         384.0 GiB\nr6i.16xlarge       memory_optimized       64         512.0 GiB\nr6i.24xlarge       memory_optimized       96         768.0 GiB\nr6i.2xlarge        memory_optimized       8          64.0 GiB\nr6i.32xlarge       memory_optimized       128        1.0 TiB\nr6i.4xlarge        memory_optimized       16         128.0 GiB\nr6i.8xlarge        memory_optimized       32         256.0 GiB\nr6id.12xlarge      memory_optimized       48         384.0 GiB\nr6id.16xlarge      memory_optimized       64         512.0 GiB\nr6id.24xlarge      memory_optimized       96         768.0 GiB\nr6id.2xlarge       memory_optimized       8          64.0 GiB\nr6id.32xlarge      memory_optimized       128        1.0 TiB\nr6id.4xlarge       memory_optimized       16         128.0 GiB\nr6id.8xlarge       memory_optimized       32         256.0 GiB\nr6id.metal         memory_optimized       128        1.0 TiB\nr6idn.12xlarge     memory_optimized       48         384.0 GiB\nr6idn.16xlarge     memory_optimized       64         512.0 GiB\nr6idn.24xlarge     memory_optimized       96         768.0 GiB\nr6idn.2xlarge      memory_optimized       8          64.0 GiB\nr6idn.32xlarge     memory_optimized       128        1.0 TiB\nr6idn.4xlarge      memory_optimized       16         128.0 GiB\nr6idn.8xlarge      memory_optimized       32         256.0 GiB\nr6idn.xlarge       memory_optimized       4          32.0 GiB\nr6id.xlarge        memory_optimized       4          32.0 GiB\nr6i.metal          memory_optimized       128        1.0 TiB\nr6in.12xlarge      memory_optimized       48         384.0 GiB\nr6in.16xlarge      memory_optimized       64         512.0 GiB\nr6in.24xlarge      memory_optimized       96         768.0 GiB\nr6in.2xlarge       memory_optimized       8          64.0 GiB\nr6in.32xlarge      memory_optimized       128        1.0 TiB\nr6in.4xlarge       memory_optimized       16         128.0 GiB\nr6in.8xlarge       memory_optimized       32         256.0 GiB\nr6in.xlarge        memory_optimized       4          32.0 GiB\nr6i.xlarge         memory_optimized       4          32.0 GiB\nr7a.12xlarge       memory_optimized       48         384.0 GiB\nr7a.16xlarge       memory_optimized       64         512.0 GiB\nr7a.24xlarge       memory_optimized       96         768.0 GiB\nr7a.2xlarge        memory_optimized       8          64.0 GiB\nr7a.32xlarge       memory_optimized       128        1.0 TiB\nr7a.48xlarge       memory_optimized       192        1.5 TiB\nr7a.4xlarge        memory_optimized       16         128.0 GiB\nr7a.8xlarge        memory_optimized       32         256.0 GiB\nr7a.metal-48xl     memory_optimized       192        1.5 TiB\nr7a.xlarge         memory_optimized       4          32.0 GiB\nr7gd.12xlarge      memory_optimized       48         384.0 GiB\nr7gd.16xlarge      memory_optimized       64         512.0 GiB\nr7gd.2xlarge       memory_optimized       8          64.0 GiB\nr7gd.4xlarge       memory_optimized       16         128.0 GiB\nr7gd.8xlarge       memory_optimized       32         256.0 GiB\nr7gd.xlarge        memory_optimized       4          32.0 GiB\nr7i.12xlarge       memory_optimized       48         384.0 GiB\nr7i.16xlarge       memory_optimized       64         512.0 GiB\nr7i.24xlarge       memory_optimized       96         768.0 GiB\nr7i.2xlarge        memory_optimized       8          64.0 GiB\nr7i.48xlarge       memory_optimized       192        1.5 TiB\nr7i.4xlarge        memory_optimized       16         128.0 GiB\nr7i.8xlarge        memory_optimized       32         256.0 GiB\nr7i.metal-24xl     memory_optimized       96         768.0 GiB\nr7i.metal-48xl     memory_optimized       192        1.5 TiB\nr7i.xlarge         memory_optimized       4          32.0 GiB\nr7iz.12xlarge      memory_optimized       48         384.0 GiB\nr7iz.16xlarge      memory_optimized       64         512.0 GiB\nr7iz.2xlarge       memory_optimized       8          64.0 GiB\nr7iz.32xlarge      memory_optimized       128        1.0 TiB\nr7iz.4xlarge       memory_optimized       16         128.0 GiB\nr7iz.8xlarge       memory_optimized       32         256.0 GiB\nr7iz.metal-16xl    memory_optimized       64         512.0 GiB\nr7iz.metal-32xl    memory_optimized       128        1.0 TiB\nr7iz.xlarge        memory_optimized       4          32.0 GiB\nu-12tb1.112xlarge  memory_optimized       448        12.0 TiB\nu-12tb1.metal      memory_optimized       448        12.0 TiB\nu-18tb1.metal      memory_optimized       448        18.0 TiB\nu-24tb1.112xlarge  memory_optimized       448        24.0 TiB\nu-24tb1.metal      memory_optimized       448        24.0 TiB\nu-3tb1.56xlarge    memory_optimized       224        3.0 TiB\nu-6tb1.112xlarge   memory_optimized       448        6.0 TiB\nu-6tb1.56xlarge    memory_optimized       224        6.0 TiB\nu-6tb1.metal       memory_optimized       448        6.0 TiB\nu-9tb1.112xlarge   memory_optimized       448        9.0 TiB\nu-9tb1.metal       memory_optimized       448        9.0 TiB\nx1.16xlarge        memory_optimized       64         976.0 GiB\nx1.32xlarge        memory_optimized       128        1.9 TiB\nx1e.16xlarge       memory_optimized       64         1.9 TiB\nx1e.2xlarge        memory_optimized       8          244.0 GiB\nx1e.32xlarge       memory_optimized       128        3.8 TiB\nx1e.4xlarge        memory_optimized       16         488.0 GiB\nx1e.8xlarge        memory_optimized       32         976.0 GiB\nx1e.xlarge         memory_optimized       4          122.0 GiB\nx2idn.16xlarge     memory_optimized       64         1.0 TiB\nx2idn.24xlarge     memory_optimized       96         1.5 TiB\nx2idn.32xlarge     memory_optimized       128        2.0 TiB\nx2idn.metal        memory_optimized       128        2.0 TiB\nx2iedn.16xlarge    memory_optimized       64         2.0 TiB\nx2iedn.24xlarge    memory_optimized       96         3.0 TiB\nx2iedn.2xlarge     memory_optimized       8          256.0 GiB\nx2iedn.32xlarge    memory_optimized       128        4.0 TiB\nx2iedn.4xlarge     memory_optimized       16         512.0 GiB\nx2iedn.8xlarge     memory_optimized       32         1.0 TiB\nx2iedn.metal       memory_optimized       128        4.0 TiB\nx2iedn.xlarge      memory_optimized       4          128.0 GiB\nx2iezn.12xlarge    memory_optimized       48         1.5 TiB\nx2iezn.2xlarge     memory_optimized       8          256.0 GiB\nx2iezn.4xlarge     memory_optimized       16         512.0 GiB\nx2iezn.6xlarge     memory_optimized       24         768.0 GiB\nx2iezn.8xlarge     memory_optimized       32         1.0 TiB\nx2iezn.metal       memory_optimized       48         1.5 TiB\nz1d.12xlarge       memory_optimized       48         384.0 GiB\nz1d.2xlarge        memory_optimized       8          64.0 GiB\nz1d.3xlarge        memory_optimized       12         96.0 GiB\nz1d.6xlarge        memory_optimized       24         192.0 GiB\nz1d.metal          memory_optimized       48         384.0 GiB\nz1d.xlarge         memory_optimized       4          32.0 GiB\nc5n.18xlarge       network_optimized      72         192.0 GiB\nc5n.2xlarge        network_optimized      8          21.0 GiB\nc5n.4xlarge        network_optimized      16         42.0 GiB\nc5n.9xlarge        network_optimized      36         96.0 GiB\nc5n.xlarge         network_optimized      4          10.5 GiB\nm5dn.12xlarge      network_optimized      48         192.0 GiB\nm5dn.16xlarge      network_optimized      64         256.0 GiB\nm5dn.24xlarge      network_optimized      96         384.0 GiB\nm5dn.2xlarge       network_optimized      8          32.0 GiB\nm5dn.4xlarge       network_optimized      16         64.0 GiB\nm5dn.8xlarge       network_optimized      32         128.0 GiB\nm5dn.xlarge        network_optimized      4          16.0 GiB\nm5n.12xlarge       network_optimized      48         192.0 GiB\nm5n.16xlarge       network_optimized      64         256.0 GiB\nm5n.24xlarge       network_optimized      96         384.0 GiB\nm5n.2xlarge        network_optimized      8          32.0 GiB\nm5n.4xlarge        network_optimized      16         64.0 GiB\nm5n.8xlarge        network_optimized      32         128.0 GiB\nm5n.xlarge         network_optimized      4          16.0 GiB\nc5ad.12xlarge      storage_optimized      48         96.0 GiB\nc5ad.16xlarge      storage_optimized      64         128.0 GiB\nc5ad.24xlarge      storage_optimized      96         192.0 GiB\nc5ad.2xlarge       storage_optimized      8          16.0 GiB\nc5ad.4xlarge       storage_optimized      16         32.0 GiB\nc5ad.8xlarge       storage_optimized      32         64.0 GiB\nc5ad.xlarge        storage_optimized      4          8.0 GiB\ni3.16xlarge        storage_optimized      64         488.0 GiB\ni3.2xlarge         storage_optimized      8          61.0 GiB\ni3.4xlarge         storage_optimized      16         122.0 GiB\ni3.8xlarge         storage_optimized      32         244.0 GiB\ni3en.12xlarge      storage_optimized      48         384.0 GiB\ni3en.24xlarge      storage_optimized      96         768.0 GiB\ni3en.2xlarge       storage_optimized      8          64.0 GiB\ni3en.3xlarge       storage_optimized      12         96.0 GiB\ni3en.6xlarge       storage_optimized      24         192.0 GiB\ni3en.metal         storage_optimized      96         768.0 GiB\ni3en.xlarge        storage_optimized      4          32.0 GiB\ni3.metal           storage_optimized      72         512.0 GiB\ni3.xlarge          storage_optimized      4          30.5 GiB\ni4i.12xlarge       storage_optimized      48         384.0 GiB\ni4i.16xlarge       storage_optimized      64         512.0 GiB\ni4i.24xlarge       storage_optimized      96         768.0 GiB\ni4i.2xlarge        storage_optimized      8          64.0 GiB\ni4i.32xlarge       storage_optimized      128        1.0 TiB\ni4i.4xlarge        storage_optimized      16         128.0 GiB\ni4i.8xlarge        storage_optimized      32         256.0 GiB\ni4i.metal          storage_optimized      128        1.0 TiB\ni4i.xlarge         storage_optimized      4          32.0 GiB\nm5ad.12xlarge      storage_optimized      48         192.0 GiB\nm5ad.16xlarge      storage_optimized      64         256.0 GiB\nm5ad.24xlarge      storage_optimized      96         384.0 GiB\nm5ad.2xlarge       storage_optimized      8          32.0 GiB\nm5ad.4xlarge       storage_optimized      16         64.0 GiB\nm5ad.8xlarge       storage_optimized      32         128.0 GiB\nm5ad.xlarge        storage_optimized      4          16.0 GiB\nm5d.12xlarge       storage_optimized      48         192.0 GiB\nm5d.16xlarge       storage_optimized      64         256.0 GiB\nm5d.24xlarge       storage_optimized      96         384.0 GiB\nm5d.2xlarge        storage_optimized      8          32.0 GiB\nm5d.4xlarge        storage_optimized      16         64.0 GiB\nm5d.8xlarge        storage_optimized      32         128.0 GiB\nm5d.xlarge         storage_optimized      4          16.0 GiB\n"
•SSSSSSS

Ran 1 of 58 Specs in 3.634 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 57 Skipped
PASS

Ginkgo ran 1 suite in 5.186322292s
Test Suite Passed

### CODE REVIEW

File: tests/e2e/test_rosacli_machine_pool.go [hunk 1]
```diff
@@ -102,6 +104,24 @@ var _ = Describe("Create machinepool",
 
 			})
 
+		It("List newly added instance-types - [id:73308]",
+			labels.Medium,
+			func() {
+				By("List the available instance-types and verify the presence of newly added instance-types")
+				newlyAddedTypes := []string{"c7a.xlarge", "c7a.48xlarge", "c7a.metal-48xl", "r7a.xlarge", "r7a.48xlarge", "r7a.metal-48xl", "hpc6a.48xlarge", "hpc6id.32xlarge", "hpc7a.96xlarge", "c7i.48xlarge", "c7i.metal-24xl", "c7i.metal-48xl", "r7i.xlarge", "r7i.48xlarge"}
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_machine_pool.go:111]: Are all these instance types supported for all regions? Will it fail when the case running on another region?

File: tests/e2e/test_rosacli_machine_pool.go [hunk 2]
```diff
@@ -102,6 +104,24 @@ var _ = Describe("Create machinepool",
...
+				availableMachineTypes, _, err := ocmResourceService.ListInstanceTypes()
+
+				if err != nil {
+					fmt.Printf("Failed to fetch instance types: %v", err)
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_machine_pool.go:115]: We don't use `fmt.Printf`. Please use log.

File: tests/e2e/test_rosacli_machine_pool.go [hunk 3]
```diff
@@ -102,6 +104,24 @@ var _ = Describe("Create machinepool",
...
+				} else {
+					var availableMachineTypesIDs []string
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_machine_pool.go:117]: There should be a common function defined in utils about the splitting.

File: tests/e2e/test_rosacli_machine_pool.go (hunk 1 above)
Comment (AkashKanni) [reply tests/e2e/test_rosacli_machine_pool.go:111]: All the instance types will not supports for all the regions, but the method `ListInstanceTypes()` lists  all the instance-types from every region, so the case wont fail when running on another region.

File: tests/e2e/test_rosacli_machine_pool.go (hunk 2 above)
Comment (AkashKanni) [reply tests/e2e/test_rosacli_machine_pool.go]: done

**Pull request merged**